Lib/test/test_tarfile.py

```diff
@@ -690,6 +691,32 @@ def format_mtime(mtime):
             tar.close()
             os_helper.rmtree(DIR)
 
+    @unittest.skipUnless(hasattr(errno, "EFTYPE"), "errno.EFTYPE required")
+    def test_extract_chmod_eftype(self):
+        # Extracting a file as non-root should skip the sticky bit (gh-108948)
+        # even on platforms where chmod fails with EFTYPE (i.e. FreeBSD). But
+        # we need to take care that any other error is preserved.
+        with ArchiveMaker() as arc:
+            arc.add("sticky", mode="?rw-rw-rwt")
+        tar = arc.open(errorlevel=2)
+        DIR = os.path.join(TEMPDIR, "chmod")
+        os.mkdir(DIR)
+        try:
+            # this should not raise:
+            tar.extract("sticky", DIR)
+            # but we can create a situation where it does raise:
+            with unittest.mock.patch("os.chmod") as mock:
+                eftype_error = OSError(errno.EFTYPE, "EFTYPE")
+                other_error = OSError(errno.EPERM, "different error")
+                mock.side_effect = [eftype_error, other_error]
+                with self.assertRaises(tarfile.ExtractError) as excinfo:
+                    tar.extract("sticky", DIR)
```

Comment:
Can you please test somewhere that the ExtractError error message is "could not change mode"?

Can you use a different name? extract("sticky", DIR) could also fail because the file already exists. Maybe add "sticky2" to the archive.